### ISSUE TITLE
Fine-Tuning the Epoch12Avg5 Model on Thai Data

### ISSUE THREAD

**Issue by lightandblue** [SpeechColab/GigaSpeech2#10]
We encountered an issue while fine-tuning your open-sourced `epoch12avg5` model. Despite the **train_loss** and **valid_loss** decreasing, we observed an **increase CER**.  Here's the setup:  

### 1. Data:  
- **Base**: Gigaspeech2 Thai data.  
- **Additional**: 3,000 hours labeled Thai audio samples.  

### 2. Changes to Training Parameters:  
Most of the model parameters are aligned with the **Zipformer-L** architecture described in the Gigaspeech2 paper. Only a few parameters were modified:  
- **Learning Rate Epoch**: `2.8 -> 3.5`  
- **Max Duration**: `1000 -> 6000`  
- **Base Learning Rate**: `0.045 -> 0.035`  
- **Warmup Batches**: `500 -> 8000`  

The training command:  
```bash  
./zipformer/train_with_bpe.py \  
  --world-size 8 \  
  --num-epochs 30 \  
  --start-epoch 1 \  
  --use-fp16 1 \  
  --exp-dir zipformer/exp1_stage02_test \  
  --max-duration 600 \  
  --lr-epochs 3.5 \  
  --base-lr 0.035 \  
  --num-encoder-layers 2,2,4,5,4,2 \  
  --downsampling-factor 1,2,4,8,4,2 \  
  --feedforward-dim 512,768,1536,2048,1536,768 \  
  --num-heads 4,4,4,8,4,4 \  
  --encoder-dim 192,256,512,768,512,256 \  
  --encoder-unmasked-dim 192,192,256,320,256,192 \  
  --query-head-dim 32 \  
  --value-head-dim 12 \  
  --pos-head-dim 4 \  
  --pos-dim 48 \  
  --cnn-module-kernel 31,31,15,15,15,31 \  
  --decoder-dim 512 \  
  --joiner-dim 512 \  
  --warmup-batches 8000 \  
  --save-every-n 2000 \  
```  

### 3. Training Details:  
- Only the model's **state_dict** was loaded.  
- The **optimizer** and **scheduler** were reset.  
![20241207-235931](https://github.com/user-attachments/assets/4dabfc33-362c-421d-9db9-5a624f8e8e20)
![20241207-235944](https://github.com/user-attachments/assets/f2b3cf1e-be99-472e-8fad-8cccd408b027)

### 4. Issue:  
Despite the **train_loss** and **valid_loss** decreasing, we observed an **increase in the Character Error Rate (CER)**.

**Comment by yfyeung** [comment]
Hi, I think such a high learning rate is not suitable for fine-tuning, and warmup should not be used, as it could lead to catastrophic forgetting. For continual training, you should use epoch-12.pt and preserve the optimizer and scheduler.

**Comment by lightandblue** [comment]
Thank you ! I will try your advice
